packages/ui/src/serializers/xml/parsers/beans-xml-parser.ts

```diff
@@ -50,12 +50,12 @@ export class BeansXmlParser {
 
     if (!constructorsElement) return undefined;
 
-    Array.from(constructorsElement.children).forEach((constructorElement) => {
+    Array.from(constructorsElement.children).forEach((constructorElement, index) => {
       const constructorIndex = constructorElement.getAttribute('index');
       const constructorValue = constructorElement.getAttribute('value');
 
-      if (constructorIndex && constructorValue) {
-        constructors = { ...constructors, [constructorIndex]: constructorValue };
+      if (constructorValue) {
+        constructors = { ...constructors, [constructorIndex ?? index]: constructorValue };
```

Comment:
this ensures that constructor is added when no index is defined and uses index from the objects forEarch.